RDA support: Make MATE's screensaver aware of being run inside a remo…

sunweaver opened the pull request:
…te desktop technology.

 This patch adds "Disconnect <remote-tech>" button to MATE's screensaver
 (if the <remote-tech> supports disconnecting from a running session).

 When this button gets clicked, the remote session will get suspended
 via librda's API (RDA = Remote Desktop Awareness).

Note on the PR: This feature is not yet fully ready for production. At least not under X2Go.

  * When resuming the session, I can take a quick glimpse at the underlying desktop session, only after a fraction of a second the screensaver dialog kicks in.
  * In fact, when resuming a MATE session via X2Go (or whatever tech that supports session resuming, my favourite approach would in fact be to unlock the session right away.
  * It may even make sense, to unlock the session directly after the session has been suspended to reduce the CPU load impact of the actual graphical screensaver (which assumingly runs in background)

Request for feedback on the above. Thanks!

sunweaver commented:
* Alternatively, a suspended session would simply blacken its screen (and hide all graphical activities from the underlying Xserver of the remote framework software)

sunweaver commented:
@monsta: @raveit65: what's the status on this PR? From IRC I read that remote desktop awareness should rather be moved into mate-session.

Basically, the mate-session code should query some parameters at session start up then, I guess, and then serve those to other MATE components.

The basic idea about all this is: MATE is nearly highly usable on remote desktop connections, even after the GTK-3 switch. For remote usage, it would be cool to switch on/off some features here and there, to make things really optimal.

This is what RDA is all about...

raveit65 commented:
@sunweaver 
As i said in irc chanel. If @monsta agree with adding code i start with testing.

sunweaver commented:
On  Mo 13 Aug 2018 18:25:13 CEST, raveit65 wrote:

> @sunweaver
> As i said in irc chanel. If @monsta agree with adding code i start  
> with testing.

@monsta: so what is your disposition on this?
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

monsta commented:
I think it's ok, but I'm not sure if multiple MATE components should all be using this rda library... I thought about handling at least some common things in the session manager.

As this currently implies changes in more than one repo, I was going to start a team discussion on this.
@sunweaver: do you have write access to our Core Team discussions?

raveit65 commented:
Doesn't matter. Installation will work without librda like before https://github.com/mate-desktop/mate-screensaver/pull/159/commits/380d759a8a801361fe9ca31f57066276326fd626#diff-67e997bcfdac55191033d57a16d1408aR438

raveit65 commented:
> do you have write access to our Core Team discussions?

I am in doubt, core-team members have access to all repos.
Why not doing a discussion in basic mate-panel PR and calling the few core members with name?
Or do we have so much developers ?

sunweaver commented:
Hi,

On  Di 14 Aug 2018 17:22:31 CEST, monsta wrote:

> Oh, and we'll need librda in distros too :slightly_smiling_face:

the patches I submitted use librda optionally. If librda is not  
available, my patchset is a no-op kind-a-thing.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
Hi,

On  Di 14 Aug 2018 17:20:17 CEST, monsta wrote:

> I think it's ok, but I'm not sure if multiple MATE components should

Nice!

> all be using this rda library... I thought about handling at least  
> some common things in the session manager.

I am not sure if this is worth the rewrite work as of now.

If RDA is generally accepted in MATE, I will write another patch for  
Caja, maybe even another caja-extenion that I will maintain in the  
future.

Also librda caches its query results (as they are not expected to  
change in the course of one login session). So the 2+ calls to librda  
will be really cheap.

> As this currently implies changes in more than one repo, I was going  
> to start a team discussion on this.
> @sunweaver: do you have write access to our Core Team discussions?

I guess, not. Please add me and highlight me (or ping me otherwise).  
(Or shall I start the discussion?)

Mike

-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
On  Di 14 Aug 2018 19:12:49 CEST, raveit65 wrote:

>> do you have write access to our Core Team discussions?
>
> I am in doubt, core-team members have access to all repos.
> Why not doing a discussion in basic mate-panel PR and calling the  
> few core members with name?
> Or do we have so much developers ?

The mate-panel PR has the better background info in its comment text.  
I agree with @raveit65.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
> This PR has been included in mate-screensaver 1.20.3-3 in Debian (and Ubuntu disco). Please test. Thanks.

So, this is independent from mate-panel PR?
And what test result is expected in x2go session?
before/ after?

raveit65 commented:
Btw. Did you fixed `make distcheck` for librda?
https://github.com/ArcticaProject/librda/issues/1#issuecomment-435595400
Otherwise i assume there is a problem with that package.

raveit65 commented:
No, you didn't fixed distcheck for librda.
```
make[4]: Entering directory '/media/Work/Programme-linux/Mate-Desktop/git-version/others/librda-0.0.2/librda-0.0.4/_build/sub/src'
  CC       librda_la-rda.lo
../../../src/rda.c:29:10: fatal error: rda_ogon.h: No such file or directory
 #include <rda_ogon.h>
          ^~~~~~~~~~~~
```

sunweaver commented:
HI,

On  Di 22 Jan 2019 11:54:41 CET, raveit65 wrote:

> No, you didn't fixed distcheck for librda.
> ```
> make[4]: Entering directory  
> '/media/Work/Programme-linux/Mate-Desktop/git-version/others/librda-0.0.2/librda-0.0.4/_build/sub/src'
>   CC       librda_la-rda.lo
> ../../../src/rda.c:29:10: fatal error: rda_ogon.h: No such file or directory
>  #include <rda_ogon.h>
>           ^~~~~~~~~~~~
> ```

This does work now with latest HEAD on  
https://github.com/ArcticaProject/librda.git

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
Hi,

> > This PR has been included in mate-screensaver 1.20.3-3 in Debian (and Ubuntu disco). Please test. Thanks.
> 
> So, this is independent from mate-panel PR?

Yes.

> And what test result is expected in x2go session?
> before/ after?

Before applying this PR: MATE screensavers in a local session and in an X2Go session look the same

After applying this PR: MATE screensaver in X2Go will offer a [ Disconnect X2Go ] button. Clicking onto it will execute an x2gosuspend-session command in the background.

sunweaver commented:
On  Fr 25 Jan 2019 10:27:04 CET, raveit65 wrote:

>> so if they don't get reports of this not working we can probably  
>> proceed to merge it.
>
> That is works in debian doesn't mean it works in other distros!
> A good example is mate-panel PR for librda which doesn't work on fedora.
> https://github.com/mate-desktop/mate-panel/pull/824#issuecomment-437574631

This should have been fixed by this RDA commit:
https://github.com/ArcticaProject/librda/commit/3f5cd65640927c562adcb4bbe6d00b0f2a0f91b4

In my test setup that I used for the demo at the X2Go Gathering, I  
assume I had an earlier version on my system where things worked.

Btw, the flaw that the above commit fixed got introduced with this commit:
https://github.com/ArcticaProject/librda/commit/d72ba23b8551ee20674c7e6e142027e77214ba1c

So, the mate-panel builds on Fedora are much likely to work now.  
However, I need to get rid of the non-RDA build's segfault. Will do  
that today.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
@sunweaver 
Can you please rebase PR against master to solve conflicts?

raveit65 commented:
And i think you can squash second commit in first one.

raveit65 commented on src/gs-lock-plug.c:
```diff
@@ -1325,6 +1377,15 @@ gs_lock_plug_class_init (GSLockPlugClass *klass)
+	g_object_class_install_property (object_class,
+	                                 PROP_SUSPEND_ENABLED,
+	                                 g_param_spec_boolean ("suspend-enabled",
+	                                         NULL,
```
Please improve Indents.

sunweaver commented:
On  Fr 25 Jan 2019 14:37:34 CET, raveit65 wrote:

> @sunweaver
> Can you please rebase PR against master to solve conflicts?

Done. The man page change of the --unlock PR sneaked into this branch.  
Cleaned up now.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
On  Fr 25 Jan 2019 15:02:54 CET, raveit65 wrote:

> raveit65 commented on this pull request.
>
>
>
>> @@ -1325,6 +1377,15 @@ gs_lock_plug_class_init (GSLockPlugClass *klass)
>  	                                         NULL,
>  	                                         FALSE,
>  	                                         G_PARAM_READWRITE));
> +#ifdef HAVE_RDA
> +	g_object_class_install_property (object_class,
> +	                                 PROP_SUSPEND_ENABLED,
> +	                                 g_param_spec_boolean ("suspend-enabled",
> +	                                         NULL,
>
> Please improve Indents.

Please look at this change in the real file, not in the Github diff  
view. Github f***s up diffs, if they contain TAB stops. IMHO,  
indenting is ok here.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
On  Fr 25 Jan 2019 14:58:39 CET, raveit65 wrote:

> And i think you can squash second commit in first one.

Done now, too.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
> Please look at this change in the real file, not in the Github diff view. Github f***s up diffs, if they contain TAB stops. IMHO, indenting is ok here.

Same with pluma here. Check tab size or better avoid to use taps

raveit65 commented:
>>> so if they don't get reports of this not working we can probably  
>>> proceed to merge it.
>>
>> That is works in debian doesn't mean it works in other distros!
>> A good example is mate-panel PR for librda which doesn't work on fedora.
>> https://github.com/mate-desktop/mate-panel/pull/824#issuecomment-437574631

>This should have been fixed by this RDA commit:
>https://github.com/ArcticaProject/librda/commit/3f5cd65640927c562adcb4bbe6d00b0f2a0f91b4

>In my test setup that I used for the demo at the X2Go Gathering, I  
>assume I had an earlier version on my system where things worked.

>Btw, the flaw that the above commit fixed got introduced with this commit:
>https://github.com/ArcticaProject/librda/commit/d72ba23b8551ee20674c7e6e142027e77214ba1c

>So, the mate-panel builds on Fedora are much likely to work now.  

I've no idea what you did test, maybe  you're kidding?

No rda-suspend button with latest librda on x2go-server session with fedora 28 and mate-1.21.x + this Pr + mate-panel rda PR.
![x2go-session-nobutton](https://user-images.githubusercontent.com/961604/51760056-ec827a00-20c9-11e9-8ac6-44b8b77df29d.jpg)

sunweaver commented:
Hi

On Friday, 25 January 2019, raveit65 wrote:
> > After applying this PR: MATE screensaver in X2Go will offer a [ Disconnect X2Go ] button. Clicking onto it will execute an x2gosuspend-session command in the background.
> 
> Where should i see this button?
> Running the command `x2gosuspend-session` in terminal will simply close the window of the x2go server. But the server is still running . So suspending the session is a bit missleading....

Suspend and resume session is X2Go terminology. I have gotten so used to it, I am too biased to find it misleading.

Suspend means: disconnect client from Xserver x2goagent and slow down the session.

Resume means: reconnect to the server and bring the suspended session back on screen, running again with normal speed.

Mike
-- 
Sent from my Sailfish device

raveit65 commented:
OK, but 

>  After applying this PR: MATE screensaver in X2Go will offer a [ Disconnect X2Go ] button. Clicking onto it will execute an x2gosuspend-session command in the background.

 Where should i see this button?

sunweaver commented:
Hi @raveit65 

I finally managed to build mate-screensaver against librda 0.0.5 on Fedora 29 using mock:

![mate-screensaver rda](https://user-images.githubusercontent.com/407231/52183132-198b0700-2805-11e9-8999-8d6c3cc64925.png)

It took me quite a while to figure that out. In general. After some trial and error, I built mate-screensaver 1.20.3 from Fedora 29 and added the RDA patch to it via Patch0: in the .spec file.

I built it inside mock. The not publicly available librda I added to the build chroot and started the next mock run with --no-clean option.

But the issue I encountered then was that the mock build would apply the patch but not show any signs of the RDA patch being applied to the configure script.

After a loooong time... my issue was: I built the modified mate-screensaver.spec from a release tarball (not from Git snapshot). And when building from release tarball, the .spec file won't call autogen.sh. Thus, no autoreconf run, either. And thus, no RDA support during configure and during build.

So... my guess now is... With your tests, you never did an autoreconf of mate-screensaver (and neither of mate-panel). Is that possible?

Thanks+Greets,
Mike
 
PS: here is the .spec file I used for mate-screensaver:

[mate-screensaver.spec.txt](https://github.com/mate-desktop/mate-screensaver/files/2826042/mate-screensaver.spec.txt)

the diff to recent mate-screensaver in Fedora is: (ignore the bit regarding %autosetup -> %setup, it was done test-wise, but not the cause)

```
[mike@minobo ~]$ diff -u mate-screensaver.spec mate-screensaver.spec.
diff: mate-screensaver.spec.: No such file or directory
[mike@minobo ~]$ diff -u mate-screensaver.spec mate-screensaver.spec.rda 
--- mate-screensaver.spec	2018-12-22 18:20:09.000000000 +0100
+++ mate-screensaver.spec.rda	2019-02-03 22:18:18.468413012 +0100
@@ -15,7 +15,7 @@
 Name:           mate-screensaver
 Version:        %{branch}.3
 %if 0%{?rel_build}
-Release:        1%{?dist}
+Release:        1.1%{?dist}
 %else
 Release:        0.9%{?git_rel}%{?dist}
 %endif
@@ -28,6 +28,7 @@
 %{?rel_build:Source0:     http://pub.mate-desktop.org/releases/%{branch}/%{name}-%{version}.tar.xz}
 # Source for snapshot-builds.
 %{!?rel_build:Source0:    http://git.mate-desktop.org/%{name}/snapshot/%{name}-%{commit}.tar.xz#/%{git_tar}}
+Patch0:         mate-screensaver-rda.patch
 
 Requires:       redhat-menus
 Requires:       system-logos
@@ -53,6 +54,7 @@
 BuildRequires:  systemd-devel
 BuildRequires:  xorg-x11-proto-devel
 BuildRequires:  xmlto
+BuildRequires:  librda-devel
 
 %description
 mate-screensaver is a screen saver and locker that aims to have
@@ -68,14 +70,16 @@
 

[librda.spec.txt](https://github.com/mate-desktop/mate-screensaver/files/2826047/librda.spec.txt)
[librda.spec.txt](https://github.com/mate-desktop/mate-screensaver/files/2826048/librda.spec.txt)
 
 %prep
-%if 0%{?rel_build}
-%autosetup -p1
-%else
-%autosetup -n %{name}-%{commit} -p1
-%endif
+# if 0%{?rel_build}
+# autosetup -p1
+# else
+# autosetup -n %{name}-%{commit} -p1
+# endif
+%setup
+%patch0 -p1 -b .rda
 
 %if 0%{?rel_build}
-#NOCONFIGURE=1 ./autogen.sh
+NOCONFIGURE=1 ./autogen.sh
 %else # 0%{?rel_build}
 # for snapshots
 # needed for git snapshots
```

PPS: and here is the patch file that I used:

[mate-screensaver-rda.patch.txt](https://github.com/mate-desktop/mate-screensaver/files/2826043/mate-screensaver-rda.patch.txt)

PPPS: and here is the librda.spec file that I used to build my librda.rpm:

[librda.spec.txt](https://github.com/mate-desktop/mate-screensaver/files/2826050/librda.spec.txt)

sunweaver commented:
When testing RDA functionality with the rdacheck utility, you should see this...

![mate rdacheck](https://user-images.githubusercontent.com/407231/52183491-aa171680-2808-11e9-9c19-7511dd6a75fb.png)

raveit65 commented:
> you never did an autoreconf of mate-screensaver (and neither of mate-panel). Is that possible?

No, i am fedora maintaner and i do always run `./autogen.sh` or `autoreconf -fi` if a patch change configure.ac or make files.
Is that new for you?
In german:
Ich fühle mich ein wenig beleidigt von Dir, nach all der Zeit die ich für deine Patches investiert habe.
Glaubst du wirklich ich hätte mehr als 6 Jahre als Fedora Maintainer überlebt, wenn meine builds an solchen Anfängerfehler scheitern würden?
.....debian ignoranZZZZZzzzzzz.....

raveit65 commented:
Part of my SPEC file:
```
# https://github.com/mate-desktop/mate-screensaver/pull/160
Patch1:         mate-screensaver_0001-gs-lock-plug-avoid-deprecated-gdk_keymap_get_default.patch
# local test
Patch2:         mate-screensaver_0001-gs-visual-gl-replace-deprecated-gdk_error_trap_push-.patch
Patch3:         mate-screensaver_0002-gs-grab-x11-replace-deprecated-gdk_error_trap_push-p.patch
# https://github.com/mate-desktop/mate-screensaver/pull/165
Patch4:         mate-screensaver_0001-lock-plug-Only-scale-lock-screen-widgets-on-older-ve.patch
# https://github.com/mate-desktop/mate-screensaver/pull/166
Patch5:         mate-screensaver_0001-Remove-monitor-pointer-from-debug-information.patch
# local test
Patch6:         mate-screensaver_0001-fade-fix-fade-computation-math.patch
# fix lockscreen
Patch7:         mate-screensaver_0001-fix-more-GdkScreen-related-deprecations.patch
# https://github.com/mate-desktop/mate-screensaver/pull/176
Patch8:         mate-screensaver_0001-mate-screensaver-command-Support-unlocking-unlock-th.patch
# https://github.com/mate-desktop/mate-screensaver/pull/159
Patch9:         mate-screensaver_0001-RDA-support-Make-MATE-s-screensaver-aware-of-being-r.patch
Patch10:        mate-screensaver_0002-configure.ac-Explicitly-require-in-RDA-0.0.3.patch

Requires:       redhat-menus
Requires:       system-logos
Requires:       gnome-keyring-pam

BuildRequires:  dbus-glib-devel
BuildRequires:  desktop-file-utils
BuildRequires:  gtk3-devel
BuildRequires:  librda-devel
BuildRequires:  libX11-devel
BuildRequires:  libXScrnSaver-devel
BuildRequires:  libXinerama-devel
BuildRequires:  libXmu-devel
BuildRequires:  libXtst-devel
BuildRequires:  libXxf86misc-devel
BuildRequires:  libXxf86vm-devel
BuildRequires:  libmatekbd-devel
BuildRequires:  libnotify-devel
BuildRequires:  mate-common
BuildRequires:  mate-desktop-devel
BuildRequires:  mate-menus-devel
BuildRequires:  mesa-libGL-devel
BuildRequires:  pam-devel
BuildRequires:  systemd-devel
BuildRequires:  xorg-x11-proto-devel
BuildRequires:  xmlto

%description
mate-screensaver is a screen saver and locker that aims to have
simple, sane, secure defaults and be well integrated with the desktop.


%package devel
Summary: Development files for mate-screensaver
Requires: %{name}%{?_isa} = %{version}-%{release}

%description devel
Development files for mate-screensaver


%prep
%if 0%{?rel_build}
%autosetup -p1
%else
%autosetup -n %{name}-%{commit} -p1
%endif

%if 0%{?rel_build}
#NOCONFIGURE=1 ./autogen.sh
%else # 0%{?rel_build}
# for snapshots
# needed for git snapshots
NOCONFIGURE=1 ./autogen.sh
%endif # 0%{?rel_build}

# for https://github.com/mate-desktop/mate-screensaver/pull/159
NOCONFIGURE=1 ./autogen.sh

%build
%configure                          \
            --with-x                \
            --disable-schemas-compile \
            --enable-docbook-docs   \
            --with-mit-ext          \
            --with-xf86gamma-ext    \
            --with-libgl            \
            --with-shadow           \
            --enable-locking        \
            --with-systemd          \
            --enable-pam            \
            --without-console-kit

make %{?_smp_mflags} V=1
```

sunweaver commented:
Hi,

On Monday, 4 February 2019, raveit65 wrote:
> Part of my SPEC file:
> ```
> # https://github.com/mate-desktop/mate-screensaver/pull/160
> Patch1:         mate-screensaver_0001-gs-lock-plug-avoid-deprecated-gdk_keymap_get_default.patch
> # local test
> Patch2:         mate-screensaver_0001-gs-visual-gl-replace-deprecated-gdk_error_trap_push-.patch
> Patch3:         mate-screensaver_0002-gs-grab-x11-replace-deprecated-gdk_error_trap_push-p.patch
> # https://github.com/mate-desktop/mate-screensaver/pull/165
> Patch4:         mate-screensaver_0001-lock-plug-Only-scale-lock-screen-widgets-on-older-ve.patch
> # https://github.com/mate-desktop/mate-screensaver/pull/166
> Patch5:         mate-screensaver_0001-Remove-monitor-pointer-from-debug-information.patch
> # local test
> Patch6:         mate-screensaver_0001-fade-fix-fade-computation-math.patch
> # fix lockscreen
> Patch7:         mate-screensaver_0001-fix-more-GdkScreen-related-deprecations.patch
> # https://github.com/mate-desktop/mate-screensaver/pull/176
> Patch8:         mate-screensaver_0001-mate-screensaver-command-Support-unlocking-unlock-th.patch
> # https://github.com/mate-desktop/mate-screensaver/pull/159
> Patch9:         mate-screensaver_0001-RDA-support-Make-MATE-s-screensaver-aware-of-being-r.patch
> Patch10:        mate-screensaver_0002-configure.ac-Explicitly-require-in-RDA-0.0.3.patch
> 
> Requires:       redhat-menus
> Requires:       system-logos
> Requires:       gnome-keyring-pam
> 
> BuildRequires:  dbus-glib-devel
> BuildRequires:  desktop-file-utils
> BuildRequires:  gtk3-devel
> BuildRequires:  librda-devel
> BuildRequires:  libX11-devel
> BuildRequires:  libXScrnSaver-devel
> BuildRequires:  libXinerama-devel
> BuildRequires:  libXmu-devel
> BuildRequires:  libXtst-devel
> BuildRequires:  libXxf86misc-devel
> BuildRequires:  libXxf86vm-devel
> BuildRequires:  libmatekbd-devel
> BuildRequires:  libnotify-devel
> BuildRequires:  mate-common
> BuildRequires:  mate-desktop-devel
> BuildRequires:  mate-menus-devel
> BuildRequires:  mesa-libGL-devel
> BuildRequires:  pam-devel
> BuildRequires:  systemd-devel
> BuildRequires:  xorg-x11-proto-devel
> BuildRequires:  xmlto
> 
> %description
> mate-screensaver is a screen saver and locker that aims to have
> simple, sane, secure defaults and be well integrated with the desktop.
> 
> 
> %package devel
> Summary: Development files for mate-screensaver
> Requires: %{name}%{?_isa} = %{version}-%{release}
> 
> %description devel
> Development files for mate-screensaver
> 
> 
> %prep
> %if 0%{?rel_build}
> %autosetup -p1
> %else
> %autosetup -n %{name}-%{commit} -p1
> %endif
> 
> %if 0%{?rel_build}
> #NOCONFIGURE=1 ./autogen.sh
> %else # 0%{?rel_build}
> # for snapshots
> # needed for git snapshots
> NOCONFIGURE=1 ./autogen.sh
> %endif # 0%{?rel_build}
> 
> # for https://github.com/mate-desktop/mate-screensaver/pull/159
> NOCONFIGURE=1 ./autogen.sh
> 
> %build
> %configure                          \
>             --with-x                \
>             --disable-schemas-compile \
>             --enable-docbook-docs   \
>             --with-mit-ext          \
>             --with-xf86gamma-ext    \
>             --with-libgl            \
>             --with-shadow           \
>             --enable-locking        \
>             --with-systemd          \
>             --enable-pam            \
>             --without-console-kit
> 
> make %{?_smp_mflags} V=1
> ```

Yes and the most important part is missing in the above spec snippet: what do you set 0%{?rel_build} to (at the very top of the spec file)? If it is set to 1, the configure file won't be recreated (as autogen.sh is commented out) and thus, RDA support won't compiled in.

Mike

-- 
Sent from my Sailfish device

raveit65 commented:
> > # for https://github.com/mate-desktop/mate-screensaver/pull/159
> > NOCONFIGURE=1 ./autogen.sh

All is done here ^^^^^^

> > 
> > %build
> > %configure                          \
> >             --with-x                \
> >             --disable-schemas-compile \
> >             --enable-docbook-docs   \
> >             --with-mit-ext          \
> >             --with-xf86gamma-ext    \
> >             --with-libgl            \
> >             --with-shadow           \
> >             --enable-locking        \
> >             --with-systemd          \
> >             --enable-pam            \
> >             --without-console-kit
> > 
> > make %{?_smp_mflags} V=1
> > ```
> 
> Yes and the most important part is missing in the above spec snippet: what do you set 0%{?rel_build}  to (at the very top of the spec file)? If it is set to 1, the configure file won't be recreated (as autogen.sh is commented out) and thus, RDA support won't compiled in.

Sadly, you don't have the skills to proper read a spec file.
Better look for another reviewer.

sunweaver commented:
Hi,

On  Mo 04 Feb 2019 09:38:47 CET, raveit65 wrote:

>> > # for https://github.com/mate-desktop/mate-screensaver/pull/159
>> > NOCONFIGURE=1 ./autogen.sh
>
> All is done here ^^^^^^

Sorry, I missed this extra line.

>> >
>> > %build
>> > %configure                          \
>> >             --with-x                \
>> >             --disable-schemas-compile \
>> >             --enable-docbook-docs   \
>> >             --with-mit-ext          \
>> >             --with-xf86gamma-ext    \
>> >             --with-libgl            \
>> >             --with-shadow           \
>> >             --enable-locking        \
>> >             --with-systemd          \
>> >             --enable-pam            \
>> >             --without-console-kit
>> >
>> > make %{?_smp_mflags} V=1
>> > ```
>>
>> Yes and the most important part is missing in the above spec  
>> snippet: what do you set 0%{?rel_build}  to (at the very top of the  
>> spec file)? If it is set to 1, the configure file won't be  
>> recreated (as autogen.sh is commented out) and thus, RDA support  
>> won't compiled in.
>
> Sadly, you don't have the skills to proper read a spec file.
> Better look for another reviewer.

I would rather get this finished with you. Please send me your  
complete mock build log (to my sunweaver@debian.org address, please).  
I will see what I can find.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
branch has been rebased against latest master + 1 more patch got added that disabled the switch user button, if session is remote.
